spec/unit/http/service/report_spec.rb

```diff
@@ -60,6 +52,15 @@
   context 'when submitting a report' do
     let(:url) { "https://www.example.com/puppet/v3/report/infinity?environment=testing" }
 
+    it 'includes puppet headers set via the :http_extra_headers and :profile settings' do
+      stub_request(:put, url).with(headers: {'Example-Header' => 'real-thing', 'X-Puppet-Profiling' => 'true'})
```

Comment:
I forgot to add a check to make sure `'another' => 'thing'` is included in the headers. I don't think it's too important, since I'm already  validating that `'Example-Header' => 'real-thing'` is added correctly, but now's the time to fix it if you think I should

Reply:
yeah let's fix that. Also I just noticed that you're testing different combinations of strings, arrays, and maps in the service tests. I think it'd be better to always pass a string in the service tests, and add a `spec/unit/settings/http_extra_headers_spec.rb` for the different ways the values can be specified.

Reply:
Ah, okay, I can do that. I knew it would be a bit confusing to have different settings formats in the different tests, but it seemed like an effective way to represent all the different cases without too many redundant tests, but I'll get that updated

Reply:
Makes sense. To avoid testing each combination of "N requests" times "M ways of setting the setting", I think it helps to test "each request includes default headers" (N of those) plus "I can set the setting as string, array, etc" (M). That way each test is only testing "one thing" and if an expectation fails later, it's clear why that is.

Reply:
That sounds like a way cleaner way of testing this. Done!